src/Lean/Elab/Tactic/Omega/Frontend.lean

```diff
@@ -526,6 +526,59 @@ def cases₂ (mvarId : MVarId) (p : Expr) (hName : Name := `h) :
     | throwError "'cases' tactic failed, unexpected new hypothesis"
   return ((s₁.mvarId, f₁), (s₂.mvarId, f₂))
 
+/--
+Helpful error message when omega cannot find a solution
+-/
+def formatErrorMessage (p : Problem) : OmegaM MessageData := do
+  if p.possible then
+    if p.isEmpty then
+      return m!"it is false"
+    else
+      let mask ← mentioned p.constraints
+      return m!"a possible counterexample may satisfy the constraints\n" ++
+        m!"{prettyConstraints p.constraints}\nwhere\n" ++
+        (← prettyAtoms mask)
+  else
+    -- formatErrorMessage should not be used in this case
+    return "it is trivially solvable"
+where
+  var (n : Nat) : String :=
+    "x" ++ ((toString (n+1)).map fun c => .ofNat (c.toNat - '0'.toNat + '₀'.toNat))
```

Comment:
Will extract that as `Nat.toSubscript`, otherwise this is probably good enough as a first iteration